src/features/decorators.ts

```diff
@@ -0,0 +1,105 @@
+import { Replace, OptionsObject } from "./feature";
+import Decimal, { DecimalSource } from "util/bignum";
+import { Computable, GetComputableType, processComputable, ProcessedComputable } from "util/computed";
+import { Persistent, State } from "game/persistence";
+import { computed, Ref, unref } from "vue";
+
+/*----====----*/
+
+export type Decorator<FeatureOptions, BaseFeature = {}, GenericFeature = {}, S extends State = State> = {
+    getPersistentData?(): Record<string, Persistent<S>>;
+    preConstruct?(feature: OptionsObject<FeatureOptions,BaseFeature,GenericFeature>): void;
+    postConstruct?(feature: OptionsObject<FeatureOptions,BaseFeature,GenericFeature>): void;
+    getGatheredProps?(feature: OptionsObject<FeatureOptions,BaseFeature,GenericFeature>): Partial<OptionsObject<FeatureOptions,BaseFeature,GenericFeature>>
+}
+
+/*----====----*/
+
+// #region Effect Decorator
```

Comment:
I think the specific decorators probably belong in their own files, or in `common.tsx` if they're particularly small and useful enough to be included as part of the "base" project (as effects would be, but maybe not the bonusAmounts/completions).